### PR TITLE
fix(docs): Configure SWA to never emit trailing slashes for website URLs

### PR DESCRIPTION
**Pull request by Josmithr**
Ensures consistent relative file path link behavior by ensuring resolved site URLs never include a trailing slash. See <https://learn.microsoft.com/en-us/azure/static-web-apps/configuration#never>

Also makes some small fixes to config files for local development.

[AB#25895](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/25895)

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../.prettierignore",
 		"rebuild": "npm run clean && npm run build",
 		"serve": "docusaurus serve",
-		"serve-with-azure-emulation": "swa start",
+		"serve-with-azure-emulation": "swa start --config-name ff-doc-site",
```
Comment (Josmithr) [inline docs/package.json:43]: Unclear from the CLI docs what the policy is for a config file with only a single "configuration" entry, but this makes things more explicit.

File: docs/static/staticwebapp.config.json [hunk 2]
```diff
@@ -14,5 +14,6 @@
 			"route": "/404",
 			"statusCode": 404
 		}
-	]
+	],
+	"trailingSlash": "never"
```
Comment (Josmithr) [inline docs/static/staticwebapp.config.json:18]: The crux of the "fix" - configures SWA to never emit a trailing slash in URLs.

File: docs/swa-cli.config.json [hunk 3]
```diff
@@ -7,10 +7,10 @@
       "appBuildCommand": "npm run build",
       "run": "npm run serve",
       "appDevserverUrl": "http://localhost:3000",
-      "swaConfigLocation": ".",
+      "swaConfigLocation": "./build",
```
Comment (Josmithr) [inline docs/swa-cli.config.json:10]:
```suggestion
      "swaConfigLocation": ".",
```

**Review by alexvy86 — APPROVED** [review]
🚀

**Pull request merged**